config/config.go

```diff
@@ -815,6 +816,9 @@ func (c *KubernetesSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 	if err := checkOverflow(c.XXX, "kubernetes_sd_config"); err != nil {
 		return err
 	}
+	if c.Kind != "nodes" && c.Kind != "service_endpoints" && c.Kind != "pods" && c.Kind != "api_servers" {
```

Comment:
Mea culpa. "pod" discovery exposes (potentially) both pods and containers. Services may expost a target that's just the service, but this is of limited value for reasons discussed in detail elsewhere (briefly, would exclude metrics for other pods that randomly provide the services; and for pods that shouhld be part of the service but are not Ready).